starter-code/pizza.js

```diff
@@ -17,69 +17,144 @@ var state = {
-  renderPepperonni()
-  renderMushrooms()
-  renderGreenPeppers()
-  renderWhiteSauce()
-  renderGlutenFreeCrust()
-
-  renderButtons()
-  renderPrice()
+  renderPepperonni();
+  renderMushrooms();
+  renderGreenPeppers();
+  renderWhiteSauce();
+  renderGlutenFreeCrust();
+
+  renderButtons();
+  renderPrice();
 }
 
 function renderPepperonni() {
-  document.querySelectorAll('.pep').forEach(function($pep){
+  document.querySelectorAll(".pep").forEach(function($pep) {
     if (state.pepperonni) {
       $pep.style.visibility = "visible";
-    }
-    else {
+    } else {
       $pep.style.visibility = "hidden";
     }
-  })
+  });
 }
 
 function renderMushrooms() {
   // Iteration 1: set the visibility of `<section class="mushroom">`
+  document.querySelectorAll(".mushroom").forEach(function($mush) {
+    if (state.mushrooms) {
+      $mush.style.visibility = "visible";
+    } else {
+      $mush.style.visibility = "hidden";
+    }
+  });
 }
 
 function renderGreenPeppers() {
   // Iteration 1: set the visibility of `<section class="green-pepper">`
+  document.querySelectorAll(".green-pepper").forEach(function($pepper) {
+    if (state.greenPeppers) {
+      $pepper.style.visibility = "visible";
+    } else {
+      $pepper.style.visibility = "hidden";
+    }
+  });
 }
 
 function renderWhiteSauce() {
   // Iteration 2: add/remove the class "sauce-white" of `<section class="sauce">`
+  document.querySelectorAll(".sauce").forEach(function($white) {
+    if (state.whiteSauce) {
+      $white.style.visibility = "visible";
+    } else {
+      $white.style.visibility = "hidden";
+    }
+  });
 }
 
 function renderGlutenFreeCrust() {
+  const glutenFree = document.querySelector(".crust");
+
+  if (state.glutenFreeCrust) {
+    glutenFree.classList.add("crust-gluten-free");
+  } else {
+    glutenFree.classList.remove("crust-gluten-free");
+  }
   // Iteration 2: add/remove the class "crust-gluten-free" of `<section class="crust">`
 }
 
+const ingredientClasses = {
+  pepperonni: "pepperonni",
+  mushrooms: "mushrooms",
+  greenPeppers: "green-peppers",
+  whiteSauce: "sauce",
+  glutenFreeCrust: "crust"
+};
+
 function renderButtons() {
-  // Iteration 3: add/remove the class "active" of each `<button class="btn">`
+  for (let item in ingredientClasses) {
+    if (state[item]) {
+      document
+        .querySelector(`.btn-${ingredientClasses[item]}`)
+        .classList.add("active");
+    } else {
+      document
+        .querySelector(".btn-" + ingredientClasses[item])
+        .classList.remove("active");
+    }
+  }
+
+  //   // Iteration 3: add/remove the class "active" of each `<button class="btn">`
+  // document.querySelectorAll("button").forEach(function(item){
+  //   item.onclick = function toggleActive(event) {
+  //     // console.log(event.currentTarget);
+  //     event.currentTarget.classList.toggle("active");
+  //   }
+  // })
 }
 
+// Iteration 4: change the HTML of `<aside class="panel price">`
 function renderPrice() {
```

Comment:
[function renderPrice() {
  // Iteration 4: change the HTML of `<aside class="panel price">`

  document.querySelector(".price ul").innerHTML = "";
  let newText = "";
  let price = basePrice;
  for (element in state) {
    if (state[element]) {
      price += ingredients[element].price;
      newText += `<li>$${ingredients[element].price} ${ingredients[element].name}</li>`;
    }
  }
  document.querySelector("strong").innerText = `$${price}`;
  document.querySelector(".price ul").innerHTML = newText;
}](url)